src/main/java/com/cognifide/knotx/repository/RepositoryConfiguration.java

```diff
@@ -0,0 +1,115 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.cognifide.knotx.repository;
+
+import com.cognifide.knotx.Server;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Configuration
+@ConfigurationProperties(locations = {"${repository.configuration}"})
+public class RepositoryConfiguration implements InitializingBean {
+
+    @Autowired
+    private Server server;
+
+    private List<RepositoryMetadata> repositories;
+
+    public Server getServer() {
+        return server;
+    }
+
+    public List<RepositoryMetadata> getRepositories() {
+        return repositories;
+    }
+
+    public void setRepositories(List<RepositoryMetadata> repositories) {
+        this.repositories = repositories;
+    }
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        String invalidMetadata = repositories.stream()
+                .filter(metadata -> !metadata.getType().validate(metadata))
+                .map(RepositoryMetadata::toString)
+                .collect(Collectors.joining(", "));
+        if (StringUtils.isNotEmpty(invalidMetadata)) {
+            throw new RuntimeException("Invalid repositories configuration " + invalidMetadata);
+        }
+    }
+
+    public static class RepositoryMetadata {
+
+        private String path;
+
+        private String serviceUrl;
+
+        private String catalogue;
+
+        private RepositoryType type;
+
+        public String getPath() {
+            return path;
+        }
+
+        public void setPath(String path) {
+            this.path = path;
+        }
+
+        public String getServiceUrl() {
+            return serviceUrl;
+        }
+
+        public void setServiceUrl(String serviceUrl) {
+            this.serviceUrl = serviceUrl;
+        }
+
+        public String getCatalogue() {
+            return catalogue;
+        }
+
+        public void setCatalogue(String catalogue) {
+            this.catalogue = catalogue;
+        }
+
+        public RepositoryType getType() {
+            return type;
+        }
+
+        public void setType(RepositoryType type) {
+            this.type = type;
+        }
+
+        @Override
+        public String toString() {
+            return "RepositoryMetadata{" +
+                    "path='" + path + '\'' +
+                    ", serviceUrl='" + serviceUrl + '\'' +
+                    ", catalogue='" + catalogue + '\'' +
+                    ", type=" + type +
+                    '}';
```

Comment:
This custom format may be a bit difficult to maintain across classes in a consistent way in the long run. I think there's already a dependency on Apache Commons, in which case we could use `ToStringBuilder`. Just a suggestion.